src/behaviours/cluster/namespaces/edit-namespace-from-new-tab.test.tsx

```diff
@@ -0,0 +1,992 @@
+        });
+
+        xit("does not show edit resource tab yet", () => {
+
+        });
+
+        describe("when clicking to edit namespace", () => {
+          beforeEach(() => {
+            // TODO: Make implementation match the description (tests above)
+            const namespaceStub = KubeObject.create(someNamespaceDataStub);
+
+            const createEditResourceTab = rendererDi.inject(createEditResourceTabInjectable);
+
+            createEditResourceTab(namespaceStub);
+          });
+
+          it("renders", () => {
+            expect(rendered.baseElement).toMatchSnapshot();
+          });
+
+          it("shows dock tab for editing namespace", () => {
+            expect(
+              rendered.getByTestId("dock-tab-for-some-first-tab-id"),
+            ).toBeInTheDocument();
+          });
+
+          it("shows spinner in the dock tab", () => {
+            expect(
+              rendered.getByTestId("edit-resource-tab-spinner"),
+            ).toBeInTheDocument();
+          });
+
+          it("calls for namespace", () => {
+            expect(callForNamespaceMock).toHaveBeenCalledWith(
+              "/apis/some-api-version/namespaces/some-uid",
+            );
+          });
+
+          describe("when call for namespace resolves with namespace", () => {
+            let someNamespace: Namespace;
+
+            beforeEach(async () => {
+              someNamespace = new Namespace({
+                apiVersion: "some-api-version",
+                kind: "Namespace",
+
+                metadata: {
+                  uid: "some-uid",
+                  name: "some-name",
+                  resourceVersion: "some-resource-version",
+                  selfLink: "/apis/some-api-version/namespaces/some-uid",
+                  somePropertyToBeRemoved: "some-value",
+                  somePropertyToBeChanged: "some-old-value",
+                },
+              });
+
+              await callForNamespaceMock.resolve({
+                callWasSuccessful: true,
+                response: someNamespace,
+              });
+            });
+
+            it("renders", () => {
+              expect(rendered.baseElement).toMatchSnapshot();
+            });
+
+            it("does not show spinner anymore", () => {
+              expect(
+                rendered.queryByTestId("edit-resource-tab-spinner"),
+              ).not.toBeInTheDocument();
+            });
+
+            it("has the configuration in editor", () => {
+              const input = rendered.getByTestId(
+                "monaco-editor-for-some-first-tab-id",
+              ) as HTMLTextAreaElement;
+
+              expect(input.value).toBe(`apiVersion: some-api-version
+kind: Namespace
+metadata:
+  uid: some-uid
+  name: some-name
+  resourceVersion: some-resource-version
+  selfLink: /apis/some-api-version/namespaces/some-uid
+  somePropertyToBeRemoved: some-value
+  somePropertyToBeChanged: some-old-value
+`);
```

Comment:
I decided not to use that because we utilize snapshots for UI. It's easy to feel "ah, there's 4 snapshots that are failing" -> check first one and see that it is valid change -> update all because you are thinking that it's the same issue on every snapshot.